bigtable/client/filters.h

```diff
@@ -0,0 +1,309 @@
+#include <chrono>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * Define the interfaces to create filter expressions.
+ *
+ * Example:
+ * @code
+ * // Get only data from the "fam" column family, and only the latest value.
+ * auto filter = Filter::Chain(Filter::Family("fam"), Filter::Latest(1));
+ * table->ReadRow("foo", std::move(filter));
+ * @endcode
+ *
+ * Those filters that use regular expressions, expect the patterns to be in
+ * the [RE2](https://github.com/google/re2/wiki/Syntax) syntax.
+ *
+ * @note Special care need be used with the expression used. Some of the
+ *   byte sequences matched (e.g. row keys, or values), can contain arbitrary
+ *   bytes, the `\C` escape sequence must be used if a true wildcard is
+ *   desired. The `.` character will not match the new line character `\n`,
+ *   which may be present in a binary value.
+ */
+class Filter {
+ public:
+  // TODO() - replace with = default if protobuf gets move constructors.
+  Filter(Filter&& rhs) noexcept : Filter() { filter_.Swap(&rhs.filter_); }
+
+  // TODO() - replace with = default if protobuf gets move constructors.
+  Filter& operator=(Filter&& rhs) noexcept {
+    Filter tmp(std::move(rhs));
+    tmp.filter_.Swap(&filter_);
+    return *this;
+  }
+
+  Filter(Filter const& rhs) = default;
+  Filter& operator=(Filter const& rhs) = default;
+
+  /// Return a filter that passes on all data.
+  static Filter PassAllFilter() {
+    Filter tmp;
+    tmp.filter_.set_pass_all_filter(true);
+    return tmp;
+  }
+
+  /// Return a filter that blocks all data.
+  static Filter BlockAllFilter() {
+    Filter tmp;
+    tmp.filter_.set_block_all_filter(true);
+    return tmp;
+  }
+
+  /// Create a filter that accepts only the last @a n values.
+  static Filter Latest(int n) {
+    Filter result;
+    result.filter_.set_cells_per_column_limit_filter(n);
+    return result;
+  }
+
+  /**
+   * Return a filter that matches column families matching the given regexp.
+   *
+   * @param pattern the regular expression.  It must be a valid
+   *     [RE2](https://github.com/google/re2/wiki/Syntax) pattern.
+   *     For technical reasons, the regex must not contain the ':' character,
+   *     even if it is not being used as a literal.
+   */
+  static Filter FamilyRegex(std::string pattern) {
+    Filter tmp;
+    tmp.filter_.set_family_name_regex_filter(std::move(pattern));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts only columns matching the given regexp.
+   *
+   * @param pattern the regular expression.  It must be a valid
+   *     [RE2](https://github.com/google/re2/wiki/Syntax) pattern.
+   */
+  static Filter ColumnRegex(std::string pattern) {
+    Filter tmp;
+    tmp.filter_.set_column_qualifier_regex_filter(std::move(pattern));
+    return tmp;
+  }
+
+  /**
+   * Create a filter that accepts columns in the given range.
+   *
+   * Notice that the range is right-open, i.e., it represents [start,end)
```

Comment:
`[start, end).`

I.e., space after comma, end with a period.

Reply:
Fixed here and a couple other locations.